### ISSUE TITLE
Update EmptyBlockCheck to Violate Empty Case Blocks with Arrow Syntax

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#15029]
child of https://github.com/checkstyle/checkstyle/issues/14961

I have read check documentation: https://checkstyle.org/checks/blocks/emptyblock.html#EmptyBlock
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> javac  src/Test.java       
 PS D:\CS\test> cat  src/Test.java         
record ColoredPoint(int p, int x, String c) { }
record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }

public class Test {

    void test(Object obj) {
        int y = 0;
        switch (obj) {
            case ColoredPoint(int x, _, _) when(x >= 2) -> { }  // expected violation
            case ColoredPoint(int x, _, _) -> { }  // expected violation
            default -> { }  // expected violation
        }
        switch (obj) {
            case ColoredPoint(int x, _, _) when(x >= 2) : { } break;  // violation
            case ColoredPoint(int x, _, _) : { } // violation
            default : { }  // violation
        }
    }
}   
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="EmptyBlockCheck">
            <property name="tokens" value="LITERAL_CASE, LITERAL_DEFAULT "/>
        </module>
    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:14:59: Must have at least one statement. [EmptyBlock]
[ERROR] D:\CS\test\src\Test.java:15:46: Must have at least one statement. [EmptyBlock]
[ERROR] D:\CS\test\src\Test.java:16:23: Must have at least one statement. [EmptyBlock]
Audit done.
Checkstyle ends with 3 errors.
PS D:\CS\test> 

```

**Describe what you want in detail**

nothing in the doc says that switch rule blocks (arrow syntax) are not validated also It can be empty so I am expecting a violation on empty blocks in the switch rule

**Comment by rnveach** [comment]
For my clarity, what is the actual difference between using `->` and `:` for the cases. Is it a delayed execution thing or ...?

Our tree calls the `->` a lambda on the case. Does this check violate on regular lambdas like this? Ex: `method(() ->{});`

**Comment by mahfouz72** [comment]
> what is the actual difference between using `->` and `:` for the cases. Is it a delayed execution thing or ...?

as far as I know, the difference is `:` Requires break statements to prevent fall-through. while with `->` no need for break statements and no fall-through. and in switch expressions `->` can be used directly to return a value without the need for braces and a local variable.


```Java
public class Test {

    void test(int n) {
        switch (n) {
            case 1 : {System.out.println("sunday");} // no break, fall through to second case
            case 2: {
                System.out.println("monday");
            }
            case 3: { } // block can be empty   // violation
        }
        switch (n) {
            case 1 -> {System.out.println("sunday");} // no fall through to second case
            case 2 -> {System.out.println("sunday");}
            case 3 -> { } // block can be empty    // expected violation

        }
        String day = switch (n) {
            case 1 -> {
                String sunday = "sunday";
                yield sunday;     // can introduce a local variable
            }
            case 2 -> "monday";  // can return value directly
            default -> {  }  // a block can't be empty (this will not compile) Switch expression should produce result in all execution paths
        };

        String day2 = switch (n) {
            case 1 : {
                String sunday = "sunday";
                yield sunday;     // must introduce a local variable
            }
            case 2 : "monday";  // can't return value directly, must introduce a local variable  (this will not compile)
            default : {  }   // a block can't be empty ( this will not compile)  // violation
        };
    }
}
```
 

> Our tree calls the `->` a lambda on the case. Does this check violate on regular lambdas like this? Ex: `method(() ->{});`

Nope. we don't support Lambda blocks in this check but for users this is just a case block, not a lambda block

**Comment by rnveach** [comment]
> the difference is : Requires break statements to prevent fall-through. while with -> no need for break statements and no fall-through. and in switch expressions -> can be used directly to return a value without the need for braces and a local variable.

Unrelated to this issue, but is this enough for a check to enforce users to use `->` or `:` where it is better appropriate? Is there any best practice here?

======

> default -> { }  // expected violation

This has no `RECORD_PATTERN_DEF`. Only a `LAMBDA`.

> case ColoredPoint(int x, _, _) when(x >= 2) : { } break;  // violation

This is covered by `LITERAL_CASE` token. Removing this token and violations go away.

> Nope. we don't support Lambda blocks in this check but for users this is just a case block, not a lambda block

I think I am good with this now.

I was a little hesitant to avoid LAMBDA since it is a LAMBDA (by our tree), but if we add LAMBDA it will cover both areas and we may not want to intertwine the 2 and may be better to have a clearer separation. The JLS (https://docs.oracle.com/javase/specs/jls/se22/html/jls-14.html#jls-14.11.1) doesn't really say LAMBDA, and the closest I found is "case label arrow". 

I think it makes sense to cover this under the tokens LITERAL_CASE and LITERAL_DEFAULT.

@nrmancuso review

**Comment by nrmancuso** [comment]
> I was a little hesitant to avoid LAMBDA since it is a LAMBDA (by our tree), but if we add LAMBDA it will cover both areas and we may not want to intertwine the 2 and may be better to have a clearer separation. The JLS (https://docs.oracle.com/javase/specs/jls/se22/html/jls-14.html#jls-14.11.1) doesn't really say LAMBDA, and the closest I found is "case label arrow".

Yes, in hindsight this token should probably not have been called a LAMBDA in our API, (I would argue that there is no "one" token that should be called this at all really) but this is a bit off topic.

This `->` token in switch statements/expressions just implies that for a given switch statement/expression, we only take exactly one branch.

I am approving this issue.
